# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/html/HTMLCanvasElement.h
# Commit: dd00e2da3a0f
# Full Hash: dd00e2da3a0f11a7cc0af9e5ece21a339f4393eb
# Author: Jeff Gilbert <jgilbert@mozilla.com>
# Date: 2020-06-11 04:04:33
# Regressor Bug: 1632249
# File Overlap Count: 3
# Description:
#   Bug 1632249 - Support compositing out-of-process WebGL. r=handyman,lsalzman,nical,geckoview-reviewers,agi,imanol
#   
#   * Majorly simplity CanvasRenderer
#   * Replace GLScreenBuffer with trivial GLSwapChain
#   * Use descriptor structs so that future SharedSurface changes aren't so painful
# ==============================================================================

diff -r 69cc1de6f754 -r dd00e2da3a0f dom/html/HTMLCanvasElement.h
--- a/dom/html/HTMLCanvasElement.h	Wed Jun 10 21:57:19 2020 +0000
+++ b/dom/html/HTMLCanvasElement.h	Wed Jun 10 22:21:02 2020 +0000
@@ -29,7 +29,6 @@
 class ClientWebGLContext;
 
 namespace layers {
-class AsyncCanvasRenderer;
 class CanvasRenderer;
 class CanvasLayer;
 class Image;
@@ -123,7 +122,6 @@
                                 public SupportsWeakPtr<HTMLCanvasElement> {
   enum { DEFAULT_CANVAS_WIDTH = 300, DEFAULT_CANVAS_HEIGHT = 150 };
 
-  typedef layers::AsyncCanvasRenderer AsyncCanvasRenderer;
   typedef layers::CanvasRenderer CanvasRenderer;
   typedef layers::CanvasLayer CanvasLayer;
   typedef layers::Layer Layer;
@@ -332,23 +330,14 @@
   layers::LayersBackend GetCompositorBackendType() const;
 
   void OnVisibilityChange();
-
   void OnMemoryPressure();
-
   void OnDeviceReset();
 
-  static void SetAttrFromAsyncCanvasRenderer(AsyncCanvasRenderer* aRenderer);
-  static void InvalidateFromAsyncCanvasRenderer(AsyncCanvasRenderer* aRenderer);
-
   already_AddRefed<layers::SharedSurfaceTextureClient> GetVRFrame();
   void ClearVRFrame();
 
   bool MaybeModified() const { return mMaybeModified; };
 
-  AsyncCanvasRenderer* GetAsyncCanvasRenderer();
-
-  layers::OOPCanvasRenderer* GetOOPCanvasRenderer();
-
  protected:
   virtual ~HTMLCanvasElement();
 
@@ -392,8 +381,7 @@
   RefPtr<HTMLCanvasPrintState> mPrintState;
   nsTArray<WeakPtr<FrameCaptureListener>> mRequestedFrameListeners;
   RefPtr<RequestedFrameRefreshObserver> mRequestedFrameRefreshObserver;
-  RefPtr<AsyncCanvasRenderer> mAsyncCanvasRenderer;
-  RefPtr<layers::OOPCanvasRenderer> mOOPCanvasRenderer;
+  RefPtr<CanvasRenderer> mCanvasRenderer;
   RefPtr<OffscreenCanvas> mOffscreenCanvas;
   RefPtr<HTMLCanvasElementObserver> mContextObserver;
 